frontend/src/components/App/Layout.tsx

```diff
@@ -56,6 +56,46 @@ function ClusterNotFoundPopup() {
 const Div = styled('div')``;
 const Main = styled('main')``;
 
+/**
+ * Merges the new cluster config with the current cluster config.
+ * It also preserves the useToken property.
+ * @param newConfig - The new cluster config.
+ * @param currentClusters - The current cluster config.
+ * @param statelessClusters - The stateless cluster config.
+ * @returns The merged cluster config.
+ */
+function mergeClusterConfigs(
+  newConfig: Omit<ConfigState, 'clusters'> & { clusters: Record<string, any> },
```

Comment:
The only thing that's used from the `newConfig` is `cluster` property, so I think it should be replaced with just `newClusters: Record<string, any>`, then responsibility of this function is now only to merge clusters.